File: swift/Examples/RelayBuilder.swift
import Foundation
import NostrSDK

private final class RejectEmptyEvents: WritePolicy {
    func admitEvent(event: Event, socketAddr: String) async -> WritePolicyResult {
        event.content().isEmpty ? .reject(message: "empty content") : .accept
    }
}

private final class LocalQueriesOnly: QueryPolicy {
    func admitQuery(query: Filter, socketAddr: String) async -> QueryPolicyResult {
        let isLocal = socketAddr.hasPrefix("127.0.0.1:") || socketAddr.hasPrefix("[::1]:")
        return isLocal ? .accept : .reject(message: "local queries only")
    }
}

func relayBuilderExample() async throws {
    let relay = LocalRelayBuilder()
        .port(port: 7676)
        .writePolicy(policy: RejectEmptyEvents())
        .queryPolicy(policy: LocalQueriesOnly())
        .build()

    try await relay.run()
    print("Relay URL: \(await relay.url())")
    try await Task.sleep(nanoseconds: UInt64.max)
}

File: python/examples/relay_builder.py
import asyncio
from nostr_sdk import *


async def main():
    relay = LocalRelayBuilder().port(7676).build()

    await relay.run()

    print(f"Relay url: {await relay.url()}")

    while True:
        await asyncio.sleep(60)


if __name__ == '__main__':
    asyncio.run(main())
